### PR TITLE
Json output

### PR DESCRIPTION
**Pull request by acheronw**

### CODE REVIEW

File: cc_corpus/corpus.py [hunk 1]
```diff
@@ -363,3 +408,32 @@ def close(self):
     def __del__(self):
         """Just calls close()."""
         self.close()
+
+
+def write_document_to_json(document: Document, output_file: typing.TextIO):
```
Comment (DavidNemeskey) [inline cc_corpus/corpus.py:413]: I would rather have a `convert_document_to_json()` function that does just the conversion and use it in-line when writing, i.e. `print(convert_document_to_json(document), file=...)`.

File: cc_corpus/corpus.py [hunk 2]
```diff
@@ -363,3 +408,32 @@ def close(self):
...
+    restructured_document = {'id': document.attrs.pop('url'),
+                             'meta': document.attrs,
+                             'text': document.content()}
+    # If we need to structure the text differently, then we will have to work
+    # with the document.paragraph attribute instead of the content() function.
+    json_document = json.dumps(restructured_document, ensure_ascii=False)
+    print(json_document, file=output_file)
+
+
+def write_file_to_json(input_file: Path, output_file: Path):
```
Comment (DavidNemeskey) [inline cc_corpus/corpus.py:431]: I would rename this to `convert_file_to_jsonl`, but otherwise OK.

**Review by DavidNemeskey — CHANGES_REQUESTED** [review]
A few minor requests. I think in hindsight, doing the `zeroe?s -> digits` conversion should have been a separate PR, but ph well.

File: scripts/frequent_paragraphs.py [hunk 3]
```diff
@@ -320,7 +320,7 @@ def read_group_documents(group: Iterator[str]) -> Iterator[Document]:
                 f = openall(doc_file, 'rb')
                 last_file = doc_file
             f.seek(int(doc_pos))
-            yield from parse(f.read(int(doc_len)).decode('utf-8').split('\n'))
+            yield from parse_docs(f.read(int(doc_len)).decode('utf-8').split('\n'))
```
[the remark this answers is not in the document]
Comment (acheronw) [reply scripts/frequent_paragraphs.py:323]: In this patch I have added jsonl support only to the renumber corpus files script.

We have five more scripts which do write documents, but not via the BatchWriter. The frequent paragraphs script is one of those. The others are: download pages, remove boilerplate, filter corpus, dedup filter.

File: cc_corpus/corpus.py (hunk 1 above)
Comment (acheronw) [reply cc_corpus/corpus.py:413]: Ok.

File: scripts/renumber_corpus_files.py [hunk 4]
```diff
@@ -22,14 +22,16 @@ def parse_arguments():
...
-    size_group = parser.add_mutually_exclusive_group()
+    size_group = parser.add_mutually_exclusive_group(required=True)
     size_group.add_argument('--documents', '-d', type=int,
                             help='the number of documents a file should contain.')
     size_group.add_argument('--keep-sizes', '-k', action='store_true',
                             help='do not merge or split files; i.e. only '
                                  'copies files to the output directory.')
-    parser.add_argument('--zeroes', '-Z', type=int, default=4,
-                        help='the number of zeroes in the output files\' names.')
+    parser.add_argument('--jsonl', '-j', action='store_true',
```
Comment (DavidNemeskey) [inline scripts/renumber_corpus_files.py:31]: I think we agreed to match the format of the output to the input. So we won't need the `-j` argument.

File: cc_corpus/corpus.py [hunk 5]
```diff
@@ -363,3 +407,31 @@ def close(self):
     def __del__(self):
         """Just calls close()."""
         self.close()
+
+
+def convert_document_to_json(document: Document):
```
Comment (DavidNemeskey) [inline cc_corpus/corpus.py:412]: I think I have found the final place for this function. Converting a `Document` to "corpus format" is done via its `__str__` method. Let's add the code below to a new `to_json` method.

**Review by DavidNemeskey — CHANGES_REQUESTED** [review]
I ran it, it works, but I would add a few last changes.

File: scripts/renumber_corpus_files.py (hunk 4 above)
Comment (acheronw) [reply scripts/renumber_corpus_files.py:31]: Ok.

File: cc_corpus/corpus.py (hunk 5 above)
Comment (acheronw) [reply cc_corpus/corpus.py:412]: Ok.

File: scripts/renumber_corpus_files.py [hunk 6]
```diff
@@ -53,7 +55,8 @@ def main():
 
     batch_size = args.documents if not args.keep_sizes else sys.maxsize
     num_docs = 0
-    with closing(BatchWriter(batch_size, args.output_dir, args.zeroes)) as bw:
+    with closing(BatchWriter(batch_size, args.output_dir, args.digits,
+                             jsonl=args.jsonl)) as bw:
         for input_file in input_files:
```
[the remark this answers is not in the document]
Comment (acheronw) [reply scripts/renumber_corpus_files.py:60]: Ok.

**Pull request merged**